### ISSUE TITLE
:TOHtml should be an external plugin/repository

### ISSUE THREAD

**Issue by bmhm** [neovim/neovim#3275]
Hello,

just thinking of core functionalities and readding issue #3000: Why would we want to have a `:TOHtml` command included in vim? A plugin can do this just as good as a core functionality.

Also, I wouldn't consider html printing of the editor's view a core functionality. Why would you want to have this feature? Either you love it or you use it ocassionaly if ever.

I know, it's just two files:
`/runtime/autoload/tohtml.vim`
and 
`runtime/syntax/2html.vim`

Still, I will be ported anyway, and changes to TOHtml should be done without haven to contribute to the main repository (for whatever reason).

Might apply to other core shipped plugins as well. Keep them in few numbers, better include sth like a plugin manager.

**Comment by Pyrohh** [comment]
On a related note, #2963 was closed partly because Nvim currently has no built-in plugin manager, so, users wanting PHPComplete support would need to use a third-party plugin manager, such as vim-plug.

With that said, I think `:TOHtml` should be an exception to that for the reasons you've given.
I can understand keeping things like netrw (needed to download spell files) and vimBall (no integrated plugin manager), but not this.  I'm biased in terms of wanting to slim down the base, so I wonder what others think. cc @fwalch @justinmk @fmoralesc

**Comment by justinmk** [comment]
I'm all for removing some of the old plugins, but not prematurely.  We need a plugin manager solution first. It's going to be awhile. What's the rush?

**Comment by fmoralesc** [comment]
Isn't this the only way to (indirectly) make "hard" copies? If so, I would make this pending on figuring out a replacement for that...

**Comment by Shougo** [comment]
If the plugin is splitted, it breaks Vim compatibility.
So, I against with the issue.

**Comment by Pyrohh** [comment]
@fmoralesc I haven't verified this, but at the top of print.txt there's this:
```
Note: If you have problems printing with |:hardcopy|, an alternative is to use
|:TOhtml| and print the resulting html file from a browser.
```

Then again, I haven't verified if `:hardcopy` works in the first place.

**Comment by fmoralesc** [comment]
That is what I meant by indirectly, indeed ;)

**Comment by bmhm** [comment]
@Shougo "If the plugin is splitted, it breaks Vim compatibility."
Not if neovim is being shipped with this plugin. I didn't say "release without this plugin", just "split plugin development".

**Comment by bmhm** [comment]
Let's put this to "on hold" until #247 is resolved.

**Comment by justinmk** [comment]
This is covered by https://github.com/neovim/neovim/issues/1507 . Feel free to make a PR after #247 is resolved, otherwise we don't need this in the open issues list.
